hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/record/MigrationContextImpl.java

```diff
@@ -48,5 +54,12 @@ public record MigrationContextImpl(
         requireNonNull(configuration);
         requireNonNull(networkInfo);
         requireNonNull(genesisRecordsBuilder);
+        requireNonNull(savepointStack);
+    }
+
+    @Override
+    public long newEntityNum() {
+        final var entityIdsFactory = new WritableStoreFactory(savepointStack, EntityIdService.NAME);
+        return entityIdsFactory.getStore(WritableEntityIdStore.class).incrementAndGet();
```

Comment:
This will create a new factory whenever somebody calls `newEntityNum()`. What do you think about creating the factory in the constructor?

Reply:
I'm fine with that. I got this code from the handle context IIRC, so I wasn't sure about the nuances

Reply:
This is no longer relevant. We aren't going to use a `WritableStoreFactory`, instead using an entity ID writable store directly, which will be passed in.